includes/RottenLinks.php

```diff
@@ -69,4 +70,27 @@ private static function getHttpStatus(
 
 		return (int)$request['code'];
 	}
+
+	/**
+	 * Get the HTTP response status for a given URL from the database.
+	 *
+	 * @param DBConnRef $dbr
+	 * @param string $url
+	 *
+	 * @return ?int null if the URL is not in the database, 0 if there was no response, or the response code
+	 */
+	public static function getResponseFromDatabase( DBConnRef $dbr, string $url ): ?int {
```

Comment:
> Also, would it be better to have 0 instead of null return then handle zero so it is always an int and don't have to be ?int? Just a thought it is fine how it is also.

There's a slight difference:
* `null`: URL not in database
* `0`: URL in database, but failed to get response
* other int: URL in database, response code

Reply:
Makes sense